wren-ui/src/components/pages/home/dashboardGrid/CacheSettingsDrawer.tsx

```diff
@@ -0,0 +1,401 @@
+                timezone: browserTimeZone,
+              }
+            : null,
+        });
+        onClose();
+      })
+      .catch(console.error);
+  };
+
+  return (
+    <Drawer
+      visible={visible}
+      title="Cache Settings"
+      width={410}
+      closable
+      destroyOnClose
+      maskClosable={false}
+      afterVisibleChange={afterVisibleChange}
+      onClose={onClose}
+      footer={
+        <Space className="d-flex justify-end">
+          <Button onClick={onClose} disabled={loading}>
+            Cancel
+          </Button>
+          <Button
+            type="primary"
+            onClick={submit}
+            loading={loading}
+            disabled={loading}
+          >
+            Submit
+          </Button>
+        </Space>
+      }
+    >
+      <Form form={form} layout="vertical">
+        <Form.Item
+          label={
+            <>
+              Caching{' '}
+              <Tooltip
+                title="Turning on caching helps dashboards load faster by avoiding redundant queries."
+                placement="right"
+              >
+                <QuestionCircleOutlined className="gray-6 ml-1 cursor-pointer" />
+              </Tooltip>
+            </>
+          }
+          name="enabled"
+          valuePropName="checked"
+          extra="Cached results will refresh automatically based on the schedule you set below."
+        >
+          <Switch />
+        </Form.Item>
+        {enabled && <Schedule />}
+      </Form>
+    </Drawer>
+  );
+}
+
+function Schedule() {
+  const form = Form.useFormInstance();
+  const frequency = Form.useWatch(['schedule', 'frequency'], form);
+  const day = Form.useWatch(['schedule', 'day'], form);
+  const time = Form.useWatch(['schedule', 'time'], form);
+  const cron = Form.useWatch(['schedule', 'cron'], form);
+
+  const onFrequencyChange = (value: string) => {
+    form.setFieldsValue({
+      schedule: { frequency: value, ...getInitialSchedule(value) },
+    });
+  };
+
+  const nextSchedule = getNextSchedule({ frequency, day, time, cron });
+
+  return (
+    <>
+      <Divider className="gray-6 text-sm">Scheduled refresh</Divider>
+      <Form.Item label="Frequency" name={['schedule', 'frequency']}>
+        <Select
+          placeholder="Select frequency"
+          options={Object.keys(FREQUENCY).map((key) => ({
+            label: capitalize(key),
+            value: FREQUENCY[key],
+          }))}
+          onChange={onFrequencyChange}
+        />
+      </Form.Item>
+
+      {frequency === FREQUENCY.DAILY && <DailyTimeSelection />}
+      {frequency === FREQUENCY.WEEKLY && <WeeklyTimeSelection />}
+      {frequency === FREQUENCY.CUSTOM && (
+        <Form.Item
+          label="Cron Expression"
+          name={['schedule', 'cron']}
+          required={false}
+          rules={[
+            {
+              required: true,
+              message: ERROR_TEXTS.CACHE_SETTINGS.CRON.REQUIRED,
+            },
+          ]}
```

Comment:
Maybe we can add validation rules to ensure that cron expressions consist primarily of 5 fields